src/inet/IPEndPointBasis.cpp

```diff
@@ -1292,20 +1293,39 @@ void IPEndPointBasis::GetPacketInfo(nw_connection_t aConnection, IPPacketInfo &
     aPacketInfo.DestPort    = nw_endpoint_get_port(dest_endpoint);
 }
 
-INET_ERROR IPEndPointBasis::GetEndPoint(nw_endpoint_t & aEndPoint, const IPAddress aAddress, uint16_t aPort)
+INET_ERROR IPEndPointBasis::GetEndPoint(nw_endpoint_t & aEndPoint, const IPAddressType aAddressType, const IPAddress aAddress,
+                                        uint16_t aPort)
 {
     INET_ERROR res = INET_NO_ERROR;
-
+    IPAddress addr;
     char addrStr[INET6_ADDRSTRLEN];
-    aAddress.ToString(addrStr, sizeof(addrStr));
-
     char portStr[INET_PORTSTRLEN];
-    snprintf(portStr, sizeof(portStr), "%u", aPort);
 
-    nw_endpoint_t endpoint = nw_endpoint_create_host(addrStr, portStr);
-    VerifyOrExit(endpoint != NULL, res = INET_ERROR_BAD_ARGS);
+    switch (aAddressType)
+    {
+
+    case kIPAddressType_IPv6:
```

Comment:
should we have a #if for the enable_ipv6 here as well?

Reply:
We could; however, CHIP mandates IPv6, so it's not clear we should unless we intend to make InetLayer an independently supportable bit of code outside of CHIP.

Reply:
Nope. I also remove this code after @gerickson comment and replace it with something clearer.